jac-uk/apply
# #1072 Working preferences questions

## What's included?

Improvements to the working preferences questions, as follows:

- [x] Working preferences questions should be mandatory (if configured)
- [x] Ranking. Should not be able to skip numbers - e.g. choosing two answers from five means user can choose 1st and 2nd choice, not 1st and 5th
- [x] There can now be multiple location questions.
- [x] There can now be multiple jurisdiction questions.
- [x] SETs can configure how many answers candidates should select (for multiple & ranked choices)
- [x] Questions can be linked together so answering one question, with a particular answer, means a follow-on question is asked
- [x] To aid navigation answers can now be grouped together
- [x] Review page should display answers

Closes #1072
Closes #1162 

## Who should test?
✅ Product owner
✅ Developers
✅ UTG

## How to test?

This should be tested alongside https://github.com/jac-uk/admin/pull/2207.

We decided to continue to support current functionality for existing exercises and have new exercises benefit from the new functionality.

There are therefore two areas to test:

**1. There should not be any breaking changes for existing exercises**

Pick an existing exercise configured with original working preferences functionality (use the existing exercise chosen/created in part 1 of testing [this PR](https://github.com/jac-uk/admin/pull/2207))

Create an application and in particular check that working preferences questions can be answered and edited.
Before submitting your application check that preference answers are correctly displayed on the Review screen. 

Please highlight any breaking changes.


**2. New exercises should benefit from the new functionality and the new functionality satisfies user requirements**

Use the new exercise created in part 2 of testing [this PR](https://github.com/jac-uk/admin/pull/2207)

Create a new application and in particular complete all of the Working preferences questions and review your answers.

Please highlight any issues with, or suggestions for, the new functionality.


## Risk - how likely is this to impact other areas?
🟠 Medium risk - this does change code that is shared with other areas

---
PREVIEW:DEVELOP
_can be OFF, DEVELOP or STAGING_
